store/sqlstore/upgrade.go

```diff
@@ -956,6 +956,7 @@ func upgradeDatabaseToVersion531(sqlStore *SqlStore) {
 	// if shouldPerformUpgrade(sqlStore, VERSION_5_30_0, VERSION_5_31_0) {
 	// allow 10 files per post
 	sqlStore.AlterColumnTypeIfExists("Posts", "FileIds", "text", "varchar(300)")
+	sqlStore.CreateColumnIfNotExists("Channels", "Shared", "tinyint(1)", "boolean", "0")
```

Comment:
Not sure if am reading this correctly, but adding a new column with a default non-null value will lock the table, because it needs an exclusive lock, since `Shared` can be null, can we just not set a default value?

Reply:
I think we should be fine since if I am reading our flow correctly, the migrations will run as the first step of an app boot up.

Reply:
Yes, good point.  Makes me wonder about the other flags added in recent versions that do set default values.

Reply:
I think so we operate under the assumption that migrations run first and then the app boots, so i guess there is no user traffic coming in to the tables while those are being modified

Reply:
Correct, however some older Postgres (including 10) will rewrite the whole table with defaults.  If this was the Posts table that could take many minutes.  Channels table includes all archived channels - it's essentially a grow only table - so some installations may get large.  We don't need a default here so I'll remove it.  Good catch.

Reply:
Yeap :100 %